geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/function/BinaryOutputEncoder.scala

```diff
@@ -59,7 +83,33 @@ object BinaryOutputEncoder extends LazyLogging {
       options: EncodingOptions,
       sort: Boolean = false): Unit = {
 
-    val sft = fc.getSchema
+    val iter = toValues(fc.getSchema, options) match {
+      case Right(toValue) => fc.features.map(toValue)
+      case Left(toValue)  => fc.features.flatMap(toValue)
+    }
+
+    // encodes the values in either 16 or 24 bytes
+    val encode: (ValuesToEncode) => Unit = options.labelField match {
+      case Some(_) => (v) => {
+        val toEncode = ExtendedValues(v.lat, v.lon, v.dtg, v.track, v.label)
+        Convert2ViewerFunction.encode(toEncode, output)
+      }
+      case None => (v) => {
+        val toEncode = BasicValues(v.lat, v.lon, v.dtg, v.track)
+        Convert2ViewerFunction.encode(toEncode, output)
+      }
+    }
+
+    if (sort) {
+      iter.toList.sorted.foreach(encode)
+    } else {
+      iter.foreach(encode)
+    }
+    // Feature collection has already been closed by SelfClosingIterator
+  }
+
+  private def toValues(sft: SimpleFeatureType, options: EncodingOptions):
+      Either[(SimpleFeature) => Seq[ValuesToEncode], (SimpleFeature) => ValuesToEncode] = {
```

Comment:
Would it be cleaner to return SF=>Seq[ValuesToEncode] here?  And then above on line 86, just flatmap over the seq?